modules/apps/portal-scheduler/portal-scheduler/src/main/java/com/liferay/portal/scheduler/internal/SchedulerEngineHelperImpl.java

```diff
@@ -997,9 +997,28 @@ public SchedulerEventMessageListener addingService(
 
 				properties.put("destination.name", destinationName);
 
-				serviceRegistration = bundleContext.registerService(
-					MessageListener.class, schedulerEventMessageListener,
-					properties);
+				if (schedulerEventMessageListener instanceof
+						SchedulerEventMessageListenerWrapper) {
+
+					serviceRegistration = bundleContext.registerService(
+						MessageListener.class, schedulerEventMessageListener,
+						properties);
+				}
+				else {
+					SchedulerEventMessageListenerWrapper
```

Comment:
@dantewang @yuhai 
The logic in else here is to replace the logic in SchedulerEngineHelper.register, right?
But the logic here is not consistent with the logic in SchedulerEngineHelper.register. 

In register(), it is checking if current registration exists, but you did the registration without any checking. Is it safe to remove the checking? Did you test the logic here?


Thanks.
Tina.